File: 4/Python/3.py
from Lenght import len

# Задача 1
print("Задача 1")
s = input("Введите слово: ")
new_s = ""

for i in s:
    new_s = i + new_s

print(new_s)

# Задача 2
print("Задача 2")
word = input("Введите слово: ")
print("*" * len(word) + word + "*" * len(word))

# Задача 3
print("\nЗадача 3")
word = input("Введите слово: ")

for i in range(2, len(word), 3):
    print(word[i])

# Задача 4
print("\nЗадача 4")
sentence = input("Введите предложение: ")
s1 = input("Введите первый символ: ")
s2 = input("Введите второй символ: ")
temp_str = ""

print(f"Вхождения символа {s1}")
for s in sentence:
    if s == " ":
        if s1 in temp_str:
            print(temp_str)
        temp_str = ""
    else:
        temp_str += s

print(f"\nВхождения символа {s2}")
for s in sentence:
    if s == " ":
        if s2 in temp_str:
            print(temp_str)
        temp_str = ""
    else:
        temp_str += s

# Задача 5
print("\nЗадача 5")
sentence = input("Введите предложение: ")
counter = 0
for s in sentence:
    if s == " ":
        counter += 1
print(counter)

# Задача 6
print("\nЗадача 6")
sentence = input("Введите предложение: ")
counter = 0

for i in range(len(sentence) - 1):
    s1 = sentence[i]
    s2 = sentence[i + 1]

    if s1 == s2:
        counter += 1

print(counter)

File: 4/Pascal/3.pas
program f3;

uses Lenght;

var
  s, new_s, word, sentence, s1, s2, temp_str: string;
  i, counter: Integer;

begin
  // Задача 1
  WriteLn('Задача 1');
  Write('Введите слово: ');
  ReadLn(s);
  new_s := '';

  for i := Length(s) downto 1 do
    new_s := new_s + s[i];

  WriteLn(new_s);

  // Задача 2
  WriteLn('Задача 2');
  Write('Введите слово: ');
  ReadLn(word);
  WriteLn(StringOfChar('*', Length(word)) + word + StringOfChar('*', Length(word)));

  // Задача 3
  WriteLn;
  WriteLn('Задача 3');
  Write('Введите слово: ');
  ReadLn(word);

  for i := 2 to Length(word) do
    if (i mod 3 = 2) then
      WriteLn(word[i]);

  // Задача 4
  WriteLn;
  writeln('Задача 4');
  write('Введите предложение: ');
  readln(sentence);
  
  write('Введите первый символ: ');
  readln(s1);
  
  write('Введите второй символ: ');
  readln(s2);
  
  writeln('Вхождения символа ', s1);
  temp_str := '';
  
  for i := 1 to len(sentence) do
  begin
    if sentence[i] = ' ' then
    begin
      if Pos(s1, temp_str) > 0 then
        writeln(temp_str);
      temp_str := '';
    end
    else
      temp_str := temp_str + sentence[i];
  end;
  
  // Проверка последнего слова
  if Pos(s1, temp_str) > 0 then
    writeln(temp_str);

  writeln;
  writeln('Вхождения символа ', s2);
  
  temp_str := '';
  
  for i := 1 to length(sentence) do
  begin
    if sentence[i] = ' ' then
    begin
      if Pos(s2, temp_str) > 0 then
        writeln(temp_str);
      temp_str := '';
    end
    else
      temp_str := temp_str + sentence[i];
  end;

  // Проверка последнего слова
  if Pos(s2, temp_str) > 0 then
    writeln(temp_str);

  // Задача 5
  WriteLn;
  WriteLn('Задача 5');
  WriteLn('Введите предложение: ');
  ReadLn(sentence);
  
  counter := 0;

  for i := 1 to len(sentence) do
  begin
    if sentence[i] = ' ' then
      counter := counter + 1;
  end;

  WriteLn('Количество пробелов: ', counter);

  // Задача 6
  WriteLn;
  WriteLn('Задача 6');
  Write('Введите предложение: ');
  ReadLn(sentence);
  counter := 0;

  for i := 1 to Length(sentence) - 1 do
    if sentence[i] = sentence[i + 1] then
      counter := counter + 1;

  WriteLn(counter);
end.

